quinn-proto/src/connection/mod.rs

```diff
@@ -748,6 +731,41 @@ where
         (buffer.len() - encode_start, pad)
     }
 
+    fn finish_and_track_packet(
```

Comment:
It could. But then it would be concerned with more than it is currently (at the moment it just builds a packet and doesn't care about congestion control - that would change). I think it's better the way it is now than bloating up the other method.

Reply:
Fair enough.